logutils/logrotation.go

```diff
@@ -20,6 +20,12 @@ type FileOptions struct {
 
 // FileHandlerWithRotation instantiates log.Handler with a configured rotation
 func FileHandlerWithRotation(opts FileOptions, format log.Format) log.Handler {
+	maxBackups := opts.MaxBackups
+	if maxBackups == 0 {
+		// Setting MaxBackups to 0 causes all log files to be kept. Even setting MaxAge to > 0 doesn't fix it
+		// Docs: https://pkg.go.dev/gopkg.in/natefinch/lumberjack.v2@v2.0.0#readme-cleaning-up-old-log-files
+		maxBackups = 1
+	}
```

Comment:
I believe it would not be advisable to override this parameter at this level of low abstraction. Instead, it should be propagated through the higher layers using the configuration:
https://github.com/status-im/status-go/blob/31144ed5a38589fd129945b634f54e800b39645a/params/config.go#L409-L410

Reply:
I also just realized I never actually used `maxBackups` :facepalm:

Reply:
Hm.. you still update it here :thinking:

Reply:
Yeah. What do you recommend? I also override it in the node_configuration in the desktop side.
I can create a migration to update the DB value of the setting, but it felt like more work for a very small fix.

Reply:
> Exactly that. Are changes here required then?

The changes here are like a fail proof. To make sure someone doesn't set 0 thinking it does no archives. There is the side effect that if someone **really** wants to keep all archives and knows the library, they might set it to 0 and then be surprised, but I doubt that would happen. Also, it's a bit crazy to want to keep all archives. A dev would set it to a high number like 10 instead IMO.

It also takes care of previous accounts, since `node_configuration` settings are only taken into account for new accounts.

Reply:
I agree, without prior knowledge of the documentation, most developers would intuitively interpret a value of '0' to signify 'no logging' rather than 'log all' within the context of node configuration.

I would personally go with:

```
diff --git a/logutils/override.go b/logutils/override.go
index 7505bca20..0d980ae6b 100644
--- a/logutils/override.go
+++ b/logutils/override.go
@@ -51,7 +51,7 @@ func OverrideRootLog(enabled bool, levelStr string, fileOpts FileOptions, termin
     var (
         handler log.Handler
     )
-    if fileOpts.Filename != "" {
+    if fileOpts.Filename != "" && fileOpts.MaxBackups > 0 {
         handler = FileHandlerWithRotation(fileOpts, log.LogfmtFormat())
     } else {
         handler = log.StreamHandler(os.Stderr, log.TerminalFormat(terminal))

```

instead of tempering with the parameter value itself.

Reply:
That would mean that all old accounts would lose the file log. I guess it kinda makes sense, since 0 means no archive :thinking: 

Also, does `log.StreamHandler` just print everything in the console? That could be crazy lol.

I did add a setting in Desktop to modify the MaxBackups value, so we can edit it in our case. It might be a bit confusing for those not aware though.

I'll try it out.

Reply:
> Also, does log.StreamHandler just print everything in the console?

Yup, that's what it does. In Debug mode, it goes wild lol.